lib/iris/coords.py

```diff
@@ -143,6 +143,147 @@ def __new__(cls, name_or_coord, minimum, maximum,
                                       'groupby_point, groupby_slice')
 
 
+def discontiguity_check_2d(coord, data, abs_tol=1e-4, transpose=False):
```

Comment:
I have a concern here about the use of an absolute tolerance, because the code will not *always* be dealing with lats+lons, and maybe not always in degrees, so you can't assume a particular value scaling.
A relative tolerance might be better, or maybe we can somehow establish a sensible value value + use a fraction of that as an absolute threshold ?